### PR TITLE
Turn some code style rule severity from suggestion to warning

### PR DESCRIPTION
**Pull request by PunkPun**
Use null propagation
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0031

Remove unnecessary parentheses
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0047-ide0048

Silencing `Use span-based 'string.Concat'` suggestion as span it is incompatible with MONO
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1845

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -138,6 +138,9 @@ dotnet_diagnostic.IDE0017.severity = warning
 # Collection initialization can be simplified
 dotnet_diagnostic.IDE0028.severity = warning
 
+# Use null propagation
```
Comment (penev92) [inline .editorconfig]: Fullstop please.

File: OpenRA.Game/Map/ActorInitializer.cs [hunk 2]
```diff
@@ -159,9 +159,8 @@ public virtual void Initialize(MiniYaml yaml)
 
 		public virtual void Initialize(T value)
 		{
-			var field = typeof(ValueActorInit<T>).GetField(nameof(value), BindingFlags.NonPublic | BindingFlags.Instance);
-			if (field != null)
-				field.SetValue(this, value);
+			typeof(ValueActorInit<T>).GetField(nameof(value), BindingFlags.NonPublic | BindingFlags.Instance)
+				?.SetValue(this, value);
```
Comment (penev92) [inline OpenRA.Game/Map/ActorInitializer.cs:164]: The line break here doesn't look right. Either do it beforen`.GetField` or all in one line. 3 Lines is also an option, but leave the '?' at the end of the line?
Same for the other two instances.

File: OpenRA.Mods.Common/Activities/RepairBridge.cs [hunk 3]
```diff
@@ -74,8 +74,8 @@ protected override void OnEnterComplete(Actor self, Actor targetActor)
 
 			if (enterLegacyHut != null)
 				enterLegacyHut.Repair(self);
-			else if (enterHut != null)
-				enterHut.Repair(self);
+			else
+				enterHut?.Repair(self);
```
Comment (penev92) [inline OpenRA.Mods.Common/Activities/RepairBridge.cs:78]: Honestly the logical thing here would be to remove the if/else entirely and just try to call both methods. Whichever isn't null will succeed.

File: OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs [hunk 4]
```diff
@@ -146,7 +146,9 @@ void IUtilityCommand.Run(Utility utility, string[] args)
 
 								data.AppendLine($"\t\t\t\tMinColor: {s.ReadUInt8():X2}{s.ReadUInt8():X2}{s.ReadUInt8():X2}");
 								data.AppendLine($"\t\t\t\tMaxColor: {s.ReadUInt8():X2}{s.ReadUInt8():X2}{s.ReadUInt8():X2}");
-								data.AppendLine($"\t\t\t\tZOffset: {(-tileSize.Height / 2.0f)}");
+								#pragma warning disable SA1021
+								data.AppendLine($"\t\t\t\tZOffset: {-tileSize.Height / 2.0f}");
+								#pragma warning restore SA1021
```
Comment (penev92) [inline OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs]: What here warrants the disabling?

**Review by penev92 — COMMENTED** [review]
Partial review

File: OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs (hunk 4 above)
Comment (PunkPun) [reply OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs]: `-` wants a space in front, but when add it then the `{` says it doesn't want a space after it

File: OpenRA.Mods.Common/Activities/RepairBridge.cs (hunk 3 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/RepairBridge.cs:78]: maybe, I haven't investigated. This just keeps the behaviour the same as before

File: .editorconfig (hunk 1 above)
Comment (PunkPun) [reply .editorconfig]: perhaps have a pass and add punctuation to all descriptions?

File: .editorconfig [hunk 5]
```diff
@@ -174,6 +183,9 @@ dotnet_diagnostic.CA1827.severity = warning
 # Use Length/Count property instead of Enumerable.Count method.
 dotnet_diagnostic.CA1829.severity = warning
 
+# Use span-based 'string.Concat' (incompatible with mono builds)
+dotnet_diagnostic.CA1845.severity = silent
```
Comment (RoosterDragon) [inline .editorconfig]: If this is a mono showstopper, let's go for 'none'.

File: OpenRA.Game/Map/ActorInitializer.cs (hunk 2 above)
Comment (RoosterDragon) [reply OpenRA.Game/Map/ActorInitializer.cs:164]: _bikeshedding_  I'd vote for 3 lines.
```c#
typeof(...)
    .GetField(...)
    ?.SetValue(...);
```

File: OpenRA.Mods.Common/Activities/RepairBridge.cs (hunk 3 above)
Comment (PunkPun) [reply OpenRA.Mods.Common/Activities/RepairBridge.cs:78]: It definitely wants to prioritise legacy hut, I don't wanna refactor the trait here

**Review by RoosterDragon — DISMISSED** [review]

File: OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs (hunk 4 above)
Comment (abcdefg30) [inline OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs]: Why not
```suggestion
								var zOffset = -tileSize.Height / 2.0f;
								data.AppendLine($"\t\t\t\tZOffset: {zOffset}");
```
?

File: OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs [hunk 6]
```diff
@@ -250,7 +250,11 @@ static unsafe int LZO1xDecompress(byte* @in, uint inLen, byte* @out, ref uint ou
 					{
 						*op++ = *ip++;
 						if (t > 2)
+						{
+							#pragma warning disable IDE0047
 							(*op++) = *ip++;
+							#pragma warning restore IDE0047
```
Comment (abcdefg30) [inline OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs:256]: Why do we need to disable this here? Are the parentheses not unnecessary but the warning still appears? Line 251 has the same? 🤔

File: OpenRA.Mods.Common/Widgets/ColorMixerWidget.cs [hunk 7]
```diff
@@ -68,7 +68,11 @@ public void SetColorLimits(float minSaturation, float maxSaturation, float v)
 					var c = (int*)cc;
 					for (var s = 0; s < 256; s++)
 					for (var h = 0; h < 256; h++)
+					{
+						#pragma warning disable IDE0047
 						(*(c + s * 256 + h)) = Color.FromAhsv(h / 255f, 1 - s / 255f, V).ToArgb();
+						#pragma warning restore IDE0047
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Widgets/ColorMixerWidget.cs:74]: Same here. Also the outer for loop would need braces.

File: OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs (hunk 6 above)
Comment (PunkPun) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs:256]: Try removing parenthesis. Something very weird is going on

File: OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs (hunk 4 above)
Comment (PunkPun) [reply OpenRA.Mods.Cnc/UtilityCommands/LegacyTilesetImporter.cs]: sure

File: OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs (hunk 6 above)
Comment (penev92) [reply OpenRA.Mods.Cnc/FileFormats/LZOCompression.cs:256]: If you remove the parenthesis rule SA1023 kicks in:
> Dereference symbol '*' should not be preceded by a space. csharp(SA1023)

File: .editorconfig [hunk 8]
```diff
@@ -183,6 +183,9 @@ dotnet_diagnostic.CA1827.severity = warning
 # Use Length/Count property instead of Enumerable.Count method.
 dotnet_diagnostic.CA1829.severity = warning
 
+# Use span-based 'string.Concat' (incompatible with mono builds).
```
Comment (penev92) [inline .editorconfig:243]: Maybe at least add a TODO here, reminding this should be enabled if/when we drop Mono?

Comment (PunkPun) [reply .editorconfig:243]: mono is mentioned

**Review by penev92 — APPROVED** [review]
👍

**Comment by penev92** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/5f6a2a7e953f810b89fefee254dfeac42f28f4f2...ce6e6487b8cf974b6e39d00e83d27d3da3488808)

**Pull request merged**